types/serde.go

```diff
@@ -17,11 +17,24 @@ func (d *Destination) UnmarshalText(text []byte) error {
 	return nil
 }
 
+type JsonRpcRequest struct {
+	Jsonrpc string      `json:"jsonrpc"`
+	Id      uint64      `json:"id"`
+	Method  string      `json:"code"`
+	Params  interface{} `json:"params"`
+}
+
+type JsonRpcResponse struct {
```

Comment:
There is also a `JsonRpcResponse` type in `rpc/serde/jsonrpc.go`. However, that one is closely tied to the specific jsonrpc method through the use of generics so I have not been able to get that one to work within the server for some of the errors (e.g. `MethodNotFoundError`). It does feel redundant to have this object defined in two places though so not sure what is the best option

Reply:
Reading https://www.jsonrpc.org/specification#response_object, it seems to me that we would like to have two types:
1. `JsonRpcSuccessResponse` (which has a result but no error)
2. `JsonRpcErrorResponse` (which has an error but no result).